src/Relations/EmbedsOneOrMany.php

```diff
@@ -392,7 +394,12 @@ public static function getUpdateValues($array, $prepend = '')
         $results = [];
 
         foreach ($array as $key => $value) {
-            $results[$prepend . $key] = $value;
+            if (str_starts_with($key, '$')) {
+                assert(is_array($value), 'Update operator value must be an array.');
+                $results[$key] = static::getUpdateValues($value, $prepend);
+            } else {
+                $results[$prepend . $key] = $value;
+            }
```

Comment:
Before, the incorrect update was:
```ts
{
  $set: {
    "address.$.updated_at": Date(...),
    "address.$.$unset": {
      "city": true,
    }
   }
}
```

After, it becomes:
```ts
{
  "$unset": {
    "address.$.city": true,
  },
  "$set": {
    "address.$.updated_at": Date(...),
  }
}
```

Field names starting with a `$` are not supported, that's why we are sure there will not be 2 levels of recursion.